pytd/query_engine.py

```diff
@@ -30,6 +30,10 @@ class QueryEngine(metaclass=abc.ABCMeta):
     """
 
     def __init__(self, apikey, endpoint, database, header):
+        if not endpoint.startswith("https://"):
+            raise ValueError(
+                "Treasure Data API server `endpoint` must start with 'https://' scheme"
```

Comment:
Treasure Data document says "endpoint" without `https://` schema https://support.treasuredata.com/hc/en-us/articles/360001474288-Sites-and-Endpoints#Endpoints
Therefore, I strongly against this validation to avoid user's confusion, rather I would think to add `https://` could be better like:

```py
if not endpoint.startswith("https://"):
    endpoint = "https://{}".format(endpoint)
```

Reply:
Sounds reasonable. 46b2c64f8ac82319939e22716389cfcb9044986e